packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1442,6 +1457,11 @@ export class Cluster extends ClusterBase {
    */
   private _openIdConnectProvider?: iam.IOpenIdConnectProvider;
 
+  /**
+   * an EKS Pod Identity Agent instance
+   */
+  private _eksPodIdentityAgent?: IAddon;
```

Comment:
for my education only, why we need both _eksPodIdentityAgent and eksPodIdentityAgent?

Reply:
`_eksPodIdentityAgent` is a private attribute for the Cluster as a state to check if this Cluster already has a eks pod identity agent AddOn created.

`eksPodIdentityAgent` is a getter method for the Cluster construct class. It essentially check if the cluster already has an AddOn created and, if not, create one for itself.